src/pp.js

```diff
@@ -360,7 +367,13 @@ function print(options, doc) {
 
                 pos = 0;
               } else {
-                out.push("\n" + _repeatIndent(ind, indentStr));
+                let indentLine;
+                if (ind in indentCache) {
+                  indentLine = indentCache[ind];
+                } else {
+                  indentCache[ind] = indentLine = _repeatIndent(ind, indentStr);
+                }
+                out.push("\n" + indentLine);
```

Comment:
Right, will do that. But the bigger issue really is the global state - if I call `prettier.format(...)` two times in the same environment with different options (I know this should be very rare, but still), the second time would use the options from the first.
That's why I put the `indentCache` object inside the `print()` function.

If the performance difference is negligible, is it better to pass the `indentCache` object to the `_repeatIndent()` function and put the check inside there?